stelaval/assignment1/SinglyLinkedList.java

```diff
@@ -0,0 +1,95 @@
+package assignment1;
+
+public class SinglyLinkedList<T> {
+
+	private Node<T> head;
+
+	public Node<T> getHead() {
+		return head;
+	}
+
+	public boolean isEmpty() {
+		return head == null;
+	}
+
+	/**
+	 * Finds the kth to last element of the singly linked list
+	 */
+	public T getKthToLastElement(int k) {
+		int length = 0;
+		Node<T> currentNode = getHead();
+
+		while (currentNode != null) {
+			currentNode = currentNode.getNext();
+			length++;
+		}
+
+		if (length < k) {
+			try {
+				throw new Exception("Index k is more than the length of the linked list");
+			} catch (Exception e) {
+				e.printStackTrace();
+			}
+		}
+
+		currentNode = getHead();
+
+		for (int i = 1; i < length - k + 1; i++)

```

Comment:
This requires traversing the list twice - once to calculate length, then second time to find the element. How could the algorithm be changed (without storing length in a field!) to traverse list only once?

Imagine that someone wants 100th element from the end of a list with 10000000000 elements.

Reply:
As per discussion during our meeting on Friday, feel free to ignore this comment.